cpp/net/connection_pool.cc

```diff
@@ -29,21 +145,59 @@ const HostPortPair& ConnectionPool::Connection::other_end() const {
 }
 
 
+void ConnectionPool::Connection::ReleaseConnection() {
+  conn_.release();
+  // Do not null out ssl_, despite the fact that it's a dangling pointer now we
+  // need it in the d'tor since the value of the pointer itself is a key into a
+  // map which needs to be updated.
```

Comment:
`conn_` *is* private, and do you mean an accessor like the existing `connection() const`?
`ReleaseConnection()` is private too, but the 'Pool has `friend` access.

What about a `Connection` which doesn't error and is just destroyed by the user? Would still need to clean up the map in the d'tor, so better to just do it in one place I think.